framework/helpers/BaseStringHelper.php

```diff
@@ -234,4 +234,26 @@ public static function endsWith($string, $with, $caseSensitive = true)
             return mb_strtolower(mb_substr($string, -$bytes, null, '8bit'), Yii::$app->charset) === mb_strtolower($with, Yii::$app->charset);
         }
     }
+
+    /**
+     * Explodes string to array, optionally trims values and skip empty
+     *
+     * @param string $string
+     * @param string $delimiter
+     * @param boolean $trim Whether to trim each element
+     * @param boolean $skipEmpty Whether to skip empty strings between delimiters
+     * @return array
+     */
+    public static function explode($string, $delimiter = ',', $trim = true, $skipEmpty = false)
+    {
+        $result = [];
+        foreach (explode($delimiter, $string) as $value) {
+            $value = $trim ? trim($value) : $value;
+            if ($skipEmpty && !static::byteLength($value)) {
```

Comment:
@samdark - yes, you're right. Replaced.
@creocoder - I thought about it yesterday, but didn't found the way to implement with `array_*` methods.

`array_map`, `array_walk` allows to change values (useful if we need to trim only), but does not allow to remove element from an array
`array_filter` allows to remove element from an array, but does not allow to change element's value

I doubt that using both of them will be faster than `foreach`. Am I missing something?